From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Vesa-Pekka Palmu <vpalmu@depili.fi>
Date: Thu, 14 Mar 2024 07:33:47 +0200
Subject: [PATCH] 6336: 6x09.sinc: Fix COM instruction

Github issue #5767 - 6x09.sinc: Fix COM instruction

The complement macro was erroneously always complementing A register
instead of the macro parameter op. This caused errors with the COMB
and COM OP1 instructions.

Closes #5767.
---
 Ghidra/Processors/MC6800/data/languages/6x09.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/MC6800/data/languages/6x09.sinc b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
index b61ef05d6e..31ef524944 100644
--- a/Ghidra/Processors/MC6800/data/languages/6x09.sinc
+++ b/Ghidra/Processors/MC6800/data/languages/6x09.sinc
@@ -430,7 +430,7 @@ macro complement(op)
 {
         $(V) = 0;
         $(C) = 1;
-        A = ~A;
+        op = ~op;
         setNZFlags(op);
 }
 
-- 
2.45.1

